railties/test/commands/credentials_test.rb

```diff
@@ -0,0 +1,34 @@
+# frozen_string_literal: true
+
+require "isolation/abstract_unit"
+require "env_helpers"
+require "rails/command"
+require "rails/commands/credentials/credentials_command"
+
+class Rails::Command::CredentialsCommandTest < ActiveSupport::TestCase
+  include ActiveSupport::Testing::Isolation, EnvHelpers
+
+  def setup
+    build_app
+  end
+
+  def teardown
+    teardown_app
+  end
+
+  test "edit command does not add master key to gitignore when already exist" do
+    run_edit_command
+
+    Dir.chdir(app_path) do
+      gitignore = File.read(".gitignore")
+      assert_equal 1, gitignore.scan(%r|config/master\.key|).length
```

Comment:
Would this have failed if it didn't have a leading slash?

Reply:
Without this fixed, `master.key` will be added to `gitignore` twice as follows. 

```
# Ignore bundler config.
/.bundle

# Ignore master key for decrypting credentials and more.
/config/master.key

# Ignore the default SQLite database.
/db/*.sqlite3
/db/*.sqlite3-journal

# Ignore all logfiles and tempfiles.
/log/*
/tmp/*
!/log/.keep
!/tmp/.keep

# Ignore uploaded files in development
/storage/*

/node_modules
/yarn-error.log

/public/assets
.byebug_history

# Ignore master key for decrypting credentials and more.
config/master.key 
``` 

As mentioned above, this test fails because there are multiple `config/master.key` states in `gitignore`

Reply:
Ah, right. Missed that it's testing there's only one entry.